app/learning.js

```diff
@@ -7,51 +7,86 @@
  * Main script for initiating the learning app.
  */
 
-define(['jquery', 'bootstrap', 'app/config', 'app/database', 'app/messages', 'app/question', 'app/timer','app/database'], function ($, bootstrap, config, db, messages, question, timer,db) {
-  timer.startTimer(".timer", config.constant("TIME_LIMIT"));
+/*jshint esversion: 6 */
+define(['jquery', 'app/lang', 'app/string', 'bootstrap', 'app/config', 'app/database', 'app/messages', 'app/question', 'app/timer', 'app/ready', 'app/user', 'app/time'], function ($, lang, string, bootstrap, config, db, messages, questions, timer, ready, user, time) {
+
+  const THOUSAND = 1000;
   var waitingForEnter = false;
-  
+
   function disableAutocomplete() {
     $('input').attr('autocomplete', 'off');
   }
-  
+
+	function inputIsEmpty() {
+		return $.trim($("#answer").val()).length === 0;
+	}
+
   function nextQuestion() {
     clearTimeout(timeout);
     timer.clearCountdown();
     messages.clear();
+    questions.nextQuestion();
     $( "#answer" ).prop("disabled", false);
     $( "#answer" ).val( "" );
     $( "#answer" ).focus();
-    question.show();
+    questions.show();
     waitingForEnter = false;
   }
-  
+
   function handleEnter() {
     if (waitingForEnter) {
       nextQuestion();
-    } else {
+    } else if (!inputIsEmpty()) {
       $( "#answer" ).prop("disabled", true);
-      question.checkAnswer();
+      questions.checkAnswer();
       waitingForEnter = true;
-      timeout = setTimeout(nextQuestion, config.constant("FEEDBACK_DELAY"));
+      timeout = setTimeout(nextQuestion,time.secondsToMilliseconds( $(".countdown").data("seconds") ) );
     }
   }
 
+  function formatFactList(items) {
+    var newList = [];
+    items.forEach(function(item) {
+      newItem = {
+        id: item.item_id,
+        text: item.item_question,
+        answer: item.item_answer,
+        hint: item.item_hint
+      };
+      newList.push(newItem);
+    });
+    return newList;
+  }
+
+	// Write localisable text to the page
+	string.fillinTextClasses();
+	$("#answer").prop("placeholder", lang("placeholder_typeanswerhere"));
+
   disableAutocomplete();
-  
-  // When the page is loaded we get the datasetId from the page url and load the dataset from the databese
-  $(document).ready(function() {
-    var url =   window.location.href
+
+	// Replace user data in view from database
+	$("span[data-replace]").each(function() {
```

Comment:
Putting it into a function will cause the npm test to complain that it is a function within a function